examples/esql-article/src/main/java/esql.article/EsqlArticle.java

```diff
@@ -0,0 +1,163 @@
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+public class EsqlArticle {
+
+    public static void main(String[] args) throws IOException, SQLException {
+        String dir = System.getProperty("user.dir");
+        Properties prop = new Properties();
+        Path path = Paths.get("src", "main","resources","application.conf");
+        prop.load(new FileInputStream(dir+ File.separator+path));
+
+        String serverUrl = prop.getProperty("server-url");
+        String apiKey = prop.getProperty("api-key");
+        String csvPath = prop.getProperty("csv-file");
+
+        RestClient restClient = RestClient
+            .builder(HttpHost.create(serverUrl))
+            .setDefaultHeaders(new Header[]{
+                new BasicHeader("Authorization", "ApiKey " + apiKey)
+            })
+            .build();
+
+        ObjectMapper mapper = JsonMapper.builder()
+            .build();
+
+        JacksonJsonpMapper jsonpMapper = new JacksonJsonpMapper(mapper);
+
+        ElasticsearchTransport transport = new RestClientTransport(
+            restClient, jsonpMapper);
+
+        ElasticsearchClient client = new ElasticsearchClient(transport);
+
+        if (!client.indices().exists(ex -> ex.index("books")).value()) {
+            client.indices()
+                .create(c -> c
+                    .index("books")
+                    .mappings(mp -> mp
+                        .properties("title", p -> p.text(t -> t))
+                        .properties("description", p -> p.text(t -> t))
+                        .properties("author", p -> p.text(t -> t))
+                        .properties("year", p -> p.short_(s -> s))
+                        .properties("publisher", p -> p.text(t -> t))
+                        .properties("ratings", p -> p.halfFloat(hf -> hf))
+                    ));
+        }
+
+        Instant start = Instant.now();
+
+        CsvMapper csvMapper = new CsvMapper();
+        CsvSchema schema = CsvSchema.builder()
+            .addColumn("title") // same order as in the csv
+            .addColumn("description")
+            .addColumn("author")
+            .addColumn("year")
+            .addColumn("publisher")
+            .addColumn("ratings")
+            .setColumnSeparator(';')
+            .setSkipFirstDataRow(true)
+            .build();
+
+        MappingIterator<Book> it = csvMapper
+            .readerFor(Book.class)
+            .with(schema)
+            .readValues(new FileReader(csvPath));
+
+        BulkIngester ingester = BulkIngester.of(bi -> bi
+            .client(client)
+            .maxConcurrentRequests(20)
+            .maxOperations(5000));
+
+        boolean hasNext = true;
+
+        while (hasNext) {
```

Comment:
Nit: use the more common `while(it.hasNext())` construct?

Reply:
it.hasNext() throws an exception if lines are malformed, I wanted to keep everything into the loop so malformed lines can be ignored